Mlem/Logic/Participating/Edit Post.swift

```diff
@@ -0,0 +1,39 @@
+//
+//  Edit Post.swift
+//  Mlem
+//
+//  Created by Sjmarf on 23/07/2023.
+//
+
+import SwiftUI
+
+// swiftlint:disable function_parameter_count
+@discardableResult func editPost(
+    postId: Int,
+    postTitle: String,
+    postBody: String,
+    postURL: String?,
+    postIsNSFW: Bool,
+    postTracker: PostTracker,
+    
+    account: SavedAccount) async throws -> APIPostView {
+        let request = EditPostRequest(
+            account: account,
+            postId: postId,
+            name: postTitle,
+            url: URL(string: postURL!),
```

Comment:
For consistency, url shouldn't be force unwrapped and instead given a default value of `""` down in EditPostRequest--this is how we handle this problem in CreatePostRequest, which isn't ideal but we should at least be consistent about it until we get a more robust (probably middleware) solution in place